File: attached_assets/zbx-chain_extracted/zbx-chain/sdk/zbx-flutter/lib/src/exceptions.dart
/// Zebvix SDK exceptions.

/// Base exception for all Zebvix SDK errors.
class ZbxException implements Exception {
  final String message;
  const ZbxException(this.message);

  @override
  String toString() => 'ZbxException: $message';
}

/// JSON-RPC error returned by the node.
class RpcException extends ZbxException {
  final int code;
  const RpcException(this.code, String message) : super(message);

  @override
  String toString() => 'RpcException($code): $message';
}

/// Wallet operation error (key derivation, signing, import).
class WalletException extends ZbxException {
  const WalletException(super.message);

  @override
  String toString() => 'WalletException: $message';
}

/// Input validation error (bad address, hash, amount).
class ValidationException extends ZbxException {
  const ValidationException(super.message);

  @override
  String toString() => 'ValidationException: $message';
}

File: attached_assets/zbx-chain_extracted/zbx-chain/sdk/zbx-python/src/zbx/exceptions.py
"""Zebvix SDK exception hierarchy."""


class ZbxError(Exception):
    """Base exception for all Zebvix SDK errors."""


class RpcError(ZbxError):
    """JSON-RPC protocol error returned by the node."""

    def __init__(self, code: int, message: str) -> None:
        self.code = code
        self.message = message
        super().__init__(f"RPC error {code}: {message}")


class WalletError(ZbxError):
    """Error in wallet operations (key derivation, signing, import)."""


class TransactionError(ZbxError):
    """Error constructing or broadcasting a transaction."""


class ValidationError(ZbxError):
    """Input validation failed (bad address, hash, amount)."""
